runtimes/runtimes/auth/auth.ts

```diff
@@ -139,6 +148,7 @@ export class Auth {
         this.connection.onNotification(bearerCredentialsDeleteNotificationType, () => {
             this.bearerCredentials = undefined
             this.connectionMetadata = undefined
+            this.lspRouter.onBearerTokenDeletion()
```

Comment:
We introduce a new pattern of using router here. Now, not only events from LSP client are routed, but also other server-interface events. Which can be ok actually, if it simplifies the implementation.